cmd/hdfs/ls.go

```diff
@@ -56,18 +58,31 @@ func ls(paths []string, long, all, humanReadable bool) {
 			}
 		}
 
-		for i, dir := range dirs {
-			if i > 0 || len(files) > 0 {
-				fmt.Println()
+		if selfOnly {
```

Comment:
Rather than switching here, I think it makes sense to just skip the step above where the dirs are separated out.

Reply:
That way `printFiles` and `printDirs` don't need to care.